waspc/cli/exe/Main.hs

```diff
@@ -98,12 +98,14 @@ main = withUtf8 . (`E.catch` handleInternalErrors) $ do
             projectName
             appDescription
             projectConfigJson
-      _ -> printWaspNewAiUsage
+      _unknownCommand -> printWaspNewAiUsage
     Command.Call.Start -> runCommand start
     Command.Call.StartDb -> runCommand Command.Start.Db.start
     Command.Call.Clean -> runCommand clean
     Command.Call.Compile -> runCommand compile
-    Command.Call.Db dbArgs -> dbCli dbArgs
+    Command.Call.Db dbArgs -> case dbArgs of
+      ["start"] -> runCommand Command.Start.Db.start
+      _dbCommand -> dbCli dbArgs
```

Comment:
The bug was introduced when I added the `reset` case into the `dbCli` by copying the line that handles `db start` and just changing the command's name.

As it turns out, all other commands inside `dbCli` perform their task on a database and rely on the same prerequisites. `wasp db start` is the only exception.

So, to prevent this bug from happening again, I pulled the `runDbCommand` call out of the case statement in `dbCli`.

The `dbCli` function is now only in charge of the commands that act on the database (I've added that to its documentation).

That said, the whole thing does look more complicated than it maybe needs to be, so I could be convinced to change it back.

@Martinsos @infomiho What do you guys think?

Reply:
+1 for using the `runDbCommand` for every db command, makes sense to me that we check prerequisites for db commands in the same way, if they don't need it, they are some other type of command.

One change I'd propose here is that this line stays like before:
```hs
Command.Call.Db dbArgs -> dbCli dbArgs
```
and you handle the `wasp db start` case above:
```hs
["start", "db"] -> Command.Call.StartDb
["db", "start"] -> Command.Call.StartDb
```

Reply:
> Or, we could just group these commands into two groups, but under dbCli: first group that applies runCommand on all of them, the other that applies runDbCommand

Can you give a code example of what you mean? I'm not sure I got it.

Reply:
You are right, it is a bit tricky: I likely didn't go enough into details when I was imagining it in my head.
You could do it I believe by doing what you did twice: having one case for normal commands, that returns Maybe, then another for those that require a DB, that also returns Maybe, and then connect those two together and if both are Nothing return `printDbUsage`. It has its drawbacks as you could by accident have same command string in both cases and now know it, although I don' think that is likely.
But since this is a bit more complicated than I imagined, it might be easier to just keep it as it was and add comments and all good.

Reply:
Ok, here's what I did in the end.

Since Miho and I prefer to have all commands under `dbCli` act on a running database, and Martin is OK with both suggestions above, I kept my change.

I also took Martin's suggestion of renaming the `runDbCommand` to something more meaningful.

The main argument is:

_If the `dbCli` function contains commands that require a database and those that don't, then its role is purely syntactic convenience (to save us from inlining all cases in main)<sup>***</sup>.
If the `dbCli` function contains only the commands that need a database, it has a meaningful semantic connection, which makes it easier to give the function a meaningful name._

That's why I renamed it from `dbCli` to `runCommandOnRunningDb`. I also changed the name of `runDbCommand` to `runCommandOnRunningDb` and used a qualified import (as we usually do).

The new name should make it obvious that all commands it runs require a running database.

I've also decided not to do this:
```haskell
["start", "db"] -> Command.Call.StartDb
["db", "start"] -> Command.Call.StartDb
```
Precisely to emphasize that, under the "db" namespace, you can run both commands that act on the database, as well as commands that don't:
```haskell
Command.Call.Db dbArgs -> case dbArgs of
  ["start"] -> runCommand Command.Start.Db.start
  _commandOnDb -> runCommandOnRunningDb dbArgs
```

To address Martin's concern:

> Yes, it has some space for the mistake that happened, but if you take runDbCommand out, one can also easily make the opposite type of mistake...

With the new names, the haddock above the function, the explicit branching under "db" in the main function, and the code path in `runCommandOnRunningDb` always going through `runDbCommand` -- I believe that both the author and the reviewer will be forced to think about whether their command needs a database running or not.

<sub>
*** You could say that the semantic connection was that they were all under the "db" namespace, but that logic was fully handled in main, and the function's call site shouldn't affect its inner semantics.
</sub>

Reply:
The new structure feels okay to me. The name is a bit weird to me: `runCommandOnRunningDb`, maybe `runCommandRequiringDb` or `runCommandWithRunningDb` or full `runCommandThatRequiresDbRunning`.

But, I'm also okay with the situation as-is.

Reply:
`runCommandThatRequiresDbRunning` - Martin suggested this one too, so I'll rename it.